monai/losses/dice.py

```diff
@@ -348,10 +348,13 @@ def forward(self, input: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
         denominator = ground_o + pred_o
 
         w = self.w_func(ground_o.float())
-        for b in w:
-            infs = torch.isinf(b)
-            b[infs] = 0.0
-            b[infs] = torch.max(b)
+        infs = torch.isinf(w)
+        if self.batch:
+            w[infs] = torch.max(w)
```

Comment:
thanks, could you help add a test case for `batch=True`, e.g.:
```py
prediction = torch.zeros(2, 3, 3, 3)
target = torch.zeros(2, 3, 3, 3)
generalized_dice_loss = GeneralizedDiceLoss(batch=True)
loss = generalized_dice_loss(prediction, target)
```

Reply:
Thanks, I have added it.

Reply:
interesting that the `loss` value is `tensor(nan, grad_fn=<MeanBackward0>)` and the backprop gradient is ok...

Reply:
> interesting that the `loss` value is `tensor(nan, grad_fn=<MeanBackward0>)` and the backprop gradient is ok...

updated to try to avoid `nan`: https://github.com/Project-MONAI/MONAI/pull/3619/commits/753b9c51c391e63282da1b964bfb2ef6aafd901b#diff-72eaf97d8b4bedaa84ed2a1053e4d520e3f56d7505f91e04844430f7d7ecfdebL353